src/systems/thermal/Thermal.cc

```diff
@@ -48,14 +51,59 @@ void Thermal::Configure(const Entity &_entity,
     gazebo::EntityComponentManager &_ecm,
     gazebo::EventManager & /*_eventMgr*/)
 {
-  if (!_sdf->HasElement("temperature"))
+  const std::string temperatureTag = "temperature";
+  const std::string heatSignatureTag = "heat_signature";
+  const std::string minTempTag = "min_temp";
+  const std::string maxTempTag = "max_temp";
+
+  if (_sdf->HasElement(temperatureTag) && _sdf->HasElement(heatSignatureTag))
+  {
+    ignerr << "Failed to load thermal system. "
+           << "Both <" << temperatureTag << "> and <" << heatSignatureTag
+           << "> were specified, but the thermal system only uses one.\n";
+  }
+  else if (_sdf->HasElement(temperatureTag))
+  {
+    double temperature = _sdf->Get<double>(temperatureTag);
+    _ecm.CreateComponent(_entity, components::Temperature(temperature));
+  }
+  else if (_sdf->HasElement(heatSignatureTag))
+  {
+    std::string heatSignature = _sdf->Get<std::string>(heatSignatureTag);
+
+    // check if this is a fuel model (if so, use the heat signature path in
+    // fuel's local cache instead of the user-specified path)
+    ignition::fuel_tools::FuelClient fuelClient;
+    std::string fuelPath;
+    if (fuelClient.CachedModelFile(ignition::common::URI(heatSignature),
+          fuelPath))
+      _ecm.CreateComponent(_entity, components::SourceFilePath(fuelPath));
+    else
+      _ecm.CreateComponent(_entity, components::SourceFilePath(heatSignature));
+
+    // see if the user defined a custom temperature range
+    float min = 0.0;
+    float max = 100.0;
+    if (_sdf->HasElement(minTempTag))
+      min = _sdf->Get<float>(minTempTag);
+    else
+      ignwarn << "Missing <" << minTempTag << ">. Defaulting to 0.\n";
```

Comment:
Add units after the `0`.

Reply:
Added in https://github.com/ignitionrobotics/ign-gazebo/pull/498/commits/a60e5d37ecc39090ce8141472f82b693c8466751